### PR TITLE
[posix] merge virtual time and real time radio

### PR DESCRIPTION
**Pull request by bukepo**
This PR merges radio implementation of virtual time and real time, so
that no need to implement twice for a new radio feature.

### CODE REVIEW

File: examples/platforms/posix/radio.c [hunk 1]
```diff
@@ -762,6 +816,16 @@ void radioTransmit(struct RadioMessage *aMessage, const struct otRadioFrame *aFr
...
+#else  // OPENTHREAD_POSIX_VIRTUAL_TIME == 0
+    struct Event event;
+
+    event.mDelay      = 1; // 1us for now
+    event.mEvent      = OT_SIM_EVENT_RADIO_RECEIVED;
+    event.mDataLength = 1 + aFrame->mLength; // include channel in first byte
+    memcpy(event.mData, aMessage, event.mDataLength);
+
+    otSimSendEvent(&event);
+#endif // OPENTHREAD_POSIX_VIRTUAL_TIME == 0
```
Comment (zhanglongxia) [inline examples/platforms/posix/radio.c]: nit: OPENTHREAD_POSIX_VIRTUAL_TIME != 0

Comment (bukepo) [reply examples/platforms/posix/radio.c]: I this this should be identical to the content after the corresponding `#if`. @jwhui

File: examples/platforms/posix/radio.c [hunk 2]
```diff
@@ -673,11 +708,32 @@ void radioSendMessage(otInstance *aInstance)
             otPlatRadioTxDone(aInstance, &sTransmitFrame, NULL, OT_ERROR_NONE);
         }
     }
+#else
+    // Wait for echo radio in virtual time mode.
+    sTxWait = true;
+#endif // OPENTHREAD_POSIX_VIRTUAL_TIME
+}
+
+bool platformRadioIsTransmitPending(void)
```
Comment (jwhui) [inline examples/platforms/posix/radio.c]: Declare as `static`?

File: examples/platforms/posix/radio.c (hunk 1 above)
Comment (jwhui) [reply examples/platforms/posix/radio.c]: Agree that the closing comment should be the same as the corresponding `#if`.

**Pull request merged**